### DISCUSSION TITLE
Introduction Letter concerning Issue Resolution Effort

### DISCUSSION THREAD

**Discussion by ata-keskin** [TUM-Dev/gocast#336] [General]
Hi all, my name is Ata Keskin. As part of my B.Sc. in Informatics I am doing a practical course on open source projects. I'd like to provide my assistance concerning the issues I have listed below. I have little knowledge in Go, but I have coded in TypeScript before and I look forward to learning the language before I go forward with the issues. I will make sure to ask for feedback where possible.

Issues I plan on working on:

Title: Stream ingest URL bugfix
Description: Stream ingest URL is not visible until reload after changing stream to selfstream
Motivation: Working on this issue will help me get familiar with the codebase and help speed up the development efforts.
Issue: https://github.com/joschahenningsen/TUM-Live/issues/236

Title: Presentation only/camera only streams
Description: Some professors only use the blackboard or don't need a camera image. They only use a subset of the three streams. On the stream administration page, theres should be an option to select camera only or presentation only. The selected stream should be streamed as default (playlist_url)
Motivation: Working on this issue will improve the quality of TUM_LIVE and help reduce unnecessary workload.
Issue: https://github.com/joschahenningsen/TUM-Live/issues/226

Title: .ics file for course
Description: Implement a tool that would be avaliable on the course pages on TUM-Live ([https://live.rbg.tum.de)](https://live.rbg.tum.de%29/), that export the date data of courses in an .ics format. 
Motivation: The ability to export this data will be useful to students for organizing their schedule.
Issue: https://github.com/joschahenningsen/TUM-Live/issues/330

Title: Combined presentation and camera view
Description: In addition to the “comb-view” video there should be an option to view both the camera display and presentation display side by side and modify their sizes. We need to make sure the videos are synchronized and at most one of them is unmuted at the same time.
Motivation: ThIs functionality will enable users to customize their display sizes, which will improve user experience.
Issue: https://github.com/joschahenningsen/TUM-Live/issues/120

**Comment by joschahenningsen** [comment]
That sounds amazing! I'll add you to our dev-chat on matrix.in.tum.de. Feel free to approach us there for help/onboarding or whatever else you might need :) 

Do you already have an issue in mind that you want to approach first?

**Reply by ata-keskin** [reply]
I plan on starting with https://github.com/joschahenningsen/TUM-Live/issues/120

I think trying to implement it will give me a good overview on the project.

**Reply by joschahenningsen** [reply]
I just noticed, that you might have a rather hard time setting this up without some demo data, I'll push a minimal sql dump with a few users later today for you to import :)

**Comment by joschahenningsen** [comment]
Hi there, Sorry for coming back to this so late. Here is a demo database you should be able to just import into a standard mariadb database: 
[tum-live-starter.zip](https://github.com/joschahenningsen/TUM-Live/files/8505487/tum-live-starter.zip)

I run mariadb in a docker container because it's very simple to set up : 
```bash
docker run --detach --name mariadb-tumlive --env MARIADB_USER=root --env MARIADB_PASSWORD=root --env MARIADB_ROOT_PASSWORD=example --restart always -p 3306:3306 mariadb:latest
```

Then I connect to the database using [DataGrip](https://www.jetbrains.com/de-de/datagrip/) (you can ofc also use mysql-client). From there its easy to create the database `tumlive` and import the sql file. 

The dump contains multiple users with different roles:

login | role
--- | --- 
admin | admin
prof1 | lecturer
prof2 | lecturer
studi1 | student
studi2 | student
studi3 | student

The password for the accounts is always `password`.
